pop_tools/regrid.py

```diff
@@ -0,0 +1,432 @@
+        )
+
+    elif grid_center_lat == 'ULAT':
+
+        # Use t-grid as corners
+        grid_corner_lat = 'TLAT'
+        grid_corner_lon = 'TLONG'
+
+        # Find grid corners
+        ds = ds.isel({'nlon': ds.nlon[:-1], 'nlat': ds.nlat[:-1]})
+
+        warnings.warn('Regridding using u-grid as cell center')
+
+    # Rename the dimensions
+    ds = ds.rename(
+        {
+            grid_center_lat: 'lat',
+            grid_center_lon: 'lon',
+            grid_corner_lat: 'lat_b',
+            grid_corner_lon: 'lon_b',
+        }
+    )
+
+    # Add the grid corner information
+    ds['lat_b'] = (['nlat_b', 'nlon_b'], corner_lats)
+    ds['lon_b'] = (['nlat_b', 'nlon_b'], corner_lons)
+
+    return ds.set_coords(['lat', 'lon', 'lat_b', 'lon_b'])
+
+
+def _prep_for_xesmf(ds, grid, lat_axis=None, cell_center='TLAT', cell_corners='ULAT'):
+    """
+    Prepares data for xesmf by - adding the grid
+    """
+
+    # add in the region mask
+    ds = _add_region_mask(ds, grid)
+
+    # Calculate the grid corners
+    if cell_center == 'TLAT':
+        corner_lats, corner_lons = gen_corner_calc(
+            ds, cell_corner_lat='ULAT', cell_corner_lon='ULONG'
+        )
+
+    else:
+        return TypeError('Does not support center coordinates other than TLAT')
+
+    # Rename the dimensions
+    ds = ds.rename({'TLAT': 'lat', 'TLONG': 'lon', 'ULAT': 'lat_b', 'ULONG': 'lon_b'})
+
+    # Add the corner lats/lons
+    ds['lat_b'] = (['nlat_b', 'nlon_b'], corner_lats)
+    ds['lon_b'] = (['nlat_b', 'nlon_b'], corner_lons)
+
+    return ds.set_coords(['lat', 'lon', 'lat_b', 'lon_b']).drop(['lat_aux_grid'])
+
+
+def gen_dest_grid(
+    method, dx=0.25, dy=0.25, lat_axis_bnds=None, lat_axis=None, lon_axis_bnds=None, lon_axis=None
+):
+    """Generates a destination grid to use with xesmf
+
+    Parameters
+    ----------
+    method: str
+       Method to use for generating the destination grid, options include 'regular_grid' or 'define_lat_aux'
+
+    dx: float default = 0.25
+       Longitudinal grid spacing to use in regular grid
+
+    dy: float default = 0.25
+       Latitudinal grid spacing to use in regular grid
+
+    lat_axis_bnds: list, array
+       Latitude axis bounds to use when defining the destination latitudes
+
+    lat_axis: list, array
+       Latitude axis to use when defining the destination latitudes
+
+    lon_axis_bounds: list, array
+       Longitude axis bounds to use when defining the destination longitudes
+
+    lon_axis: list, array
+       Longitude axis to use when defining the destination longitudes
+
+    Returns
+    -------
+    xarray.Dataset
+      An xarray.Dataset with the associated coordinates, formatted with lat, lon, lat_b, lon_b
+    """
+
+    reg_grid = xe.util.grid_global(dx, dy)
+
+    if method == 'regular_grid':
+        out_ds = reg_grid
+
+    elif method == 'lat_axis':
+        out_ds = xr.Dataset()
+
+        if lat_axis_bnds is not None:
```

Comment:
It appears that `lat_axis_bnds` is required for this function to properly work. Instead of having it as a keyword argument, could we just make it a required argument and remove this check?